api/handlers/devicehandler.py

```diff
@@ -0,0 +1,124 @@
+import datetime as dt
+
+from .. import base
+from .. import config
+from .. import util
+from ..dao import containerstorage, APIPermissionException, APINotFoundException
+from ..types import Origin
+
+log = config.log
+
+Status = util.Enum('Origin', {
+    'ok':       'ok',       # Device's last seen time is shorter than expected interval for checkin, no errors listed.
+    'missing':  'missing',  # Device's last seen time is longer than the expected interval for checkin, but no errors listed.
+    'error':    'error' ,   # Device has errors listed.
+    'unknown':  'unknown'   # Device did not set an expected checkin interval.
+})
+
+
```

Comment:
Let me know what you think of these simple permissions checking decorators. I thought I'd experiment with this change and if we like them, I'll move them to the auth package instead.